### PR TITLE
Add empty option to drop down widget

### LINKED ISSUE #940: Add an empty option for DropDown Widget
**Is your feature request related to a problem? Please describe.**
Error correction for the [DropDown Widget](https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt) 

**Describe the solution you'd like**
I was planning to have something similar to #781 so that clicking on the drop-down choice would clear the answer.

**Describe alternatives you've considered**

**Additional context**
[ Choice questions should have clear options](https://github.com/google/android-fhir/pull/546#issuecomment-872723271
)

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by maanuanubhav999**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #940 

**Description**
add an empty option for the dropdown widget.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature 

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by maanuanubhav999** [comment]
Hey @jingtang10, can you give me some insights on how can I write tests cases for this condition since I tried many methods but they were not working?

## Goal
I have to click on the dropdown and select the option for the test case but I couldn't find any method to do the same.

## Thing I tried
* using setListSelection even when the pop-up is visible and the list is not empty.
* using setText but it will not invoke the click listener.
* Also looked into the source code but nothing that could help.

Something like this could easily be handled with espresso but should I use it? thanks.

**Comment by Tarun-Bhardwaj** [comment]
@maanuanubhav999 , did you get a clarity on the approach or you need help on this?
CC: @jingtang10

**Comment by maanuanubhav999** [comment]
Hey, @Tarun-Bhardwaj  I would like some help with the writing test.  I am not able to find some direct method to do so.

**Comment by Tarun-Bhardwaj** [comment]
> Hey, @Tarun-Bhardwaj I would like some help with the writing test. I am not able to find some direct method to do so.

@anita-vadverao , is this something you could help Anubhav with?

CC: @jingtang10

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt [hunk 1]
```diff
@@ -59,7 +59,8 @@ internal object QuestionnaireItemDropDownViewHolderFactory :
         }
         textView.text = questionnaireItemViewItem.questionnaireItem.localizedText
         val answerOptionString =
-          this.questionnaireItemViewItem.answerOption.map { it.displayString }
+          this.questionnaireItemViewItem.answerOption.map { it.displayString }.toMutableList()
+        answerOptionString.add(0, "-Select-")
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:63]: Can you please add an empty string?

If we add something like "-Select-" we'd have to translate it to different langauges

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt (hunk 1 above)
Comment (maanuanubhav999) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:63]: sure

**Comment by jingtang10** [comment]
Have you tried using `performClick`? search for it in our codebase.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt (hunk 1 above)
Comment (anita-vadverao) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:63]: > > Hey, @Tarun-Bhardwaj I would like some help with the writing test. I am not able to find some direct method to do so.
> 
> @anita-vadverao , is this something you could help Anubhav with?
> 
> CC: @jingtang10

@maanuanubhav999 Did you try with `performClick` with `getChildAt()` to select item from drop-down?

Comment (maanuanubhav999) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:63]: @jingtang10 @anita-vadverao thanks for the response.

 I did try `performClick` with `getChildAt()` with AutoCompleteTextView but was unable to simulate a click on the dropdown.
 
 with AutoCompleteTextView the `getChildAt()` method does not exists.

Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:63]: does this help: https://www.examplefiles.net/cs/821426

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Comment by jingtang10** [comment]
Any update @maanuanubhav999 ?

**Comment by maanuanubhav999** [comment]
sorry @jingtang10 didn't get much time this week, will update by tomorrow for sure.

**Comment by jingtang10** [comment]
any update on this?

**Comment by maanuanubhav999** [comment]
@jingtang10 Thanks for your patience.

I did try running the espresso test but the test seems to be running forever, 
When I tried debugging it, finding the view could be archived but after performing click on it the weird running forever scenario would happen again.

So I am still working on it.

**Comment by jingtang10** [comment]
> @jingtang10 Thanks for your patience.
> 
> I did try running the espresso test but the test seems to be running forever, 
> When I tried debugging it, finding the view could be archived but after performing click on it the weird running forever scenario would happen again.
> 
> So I am still working on it. 

Thanks for the update. If you want others to be able to help you need to push your commits so others can see.

**Comment by jingtang10** [comment]
@vvikramraj is starting to work on some UI tests so maybe he will be able to help (or this will help him).

@anita-vadverao fyi also

**Comment by maanuanubhav999** [comment]
thanks @jingtang10

**Comment by jingtang10** [comment]
@epicadk @PallaviGanorkar are you able to take a look and help with this? thanks

**Comment by jingtang10** [comment]
please see #1298 from @PallaviGanorkar

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryInstrumentedTest.kt [hunk 2]
```diff
@@ -117,7 +117,7 @@ class QuestionnaireItemDropDownViewHolderFactoryInstrumentedTest {
           .itemView
           .findViewById<AutoCompleteTextView>(R.id.auto_complete)
           .adapter
-          .getItem(0)
+          .getItem(1)
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryInstrumentedTest.kt:120]: you need another test case to check that the empty option is populated in the ui

**Review by jingtang10 — APPROVED** [review]
thanks @maanuanubhav999 for this change. approving since @PallaviGanorkar is addressing the tests in her PR #1298

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryInstrumentedTest.kt (hunk 2 above)
Comment (maanuanubhav999) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryInstrumentedTest.kt:120]: Hey @jingtang10, thanks for being patience with me, got to learn a lot.
